### ISSUE TITLE
cancelCompression is not working

### ISSUE THREAD

**Issue by Arley011** [jonataslaw/VideoCompress#164]
When I call VideoCompress.cancelCompression() my progress stream stops receiving new values, but VideoCompress.compressVideo()  doesn't return anything. Seems like compressing is still in progress cuz after some time compressVideo() will return MediaInfo with compressed file (and with isCanceled = true). 
I even tried to use `CancelableOperation.fromFuture(VideoCompress.compressVideo())` and it works for me, but then I noticed that if I will select the next video to compress right after cancelation, then the compression progress stream shows 0. And after some time (seems like it waits for the previous compression to complete even if I canceled it) stream will produce valid progress values.

So, please, take a look at the cancel method. I've tested on iPhone 12 with iOS 15.2.1.

**Comment by jpetro416** [comment]
So does catching the error like this and canceling compression work?
```

 Future<File> _compressVideo(File file) async {
    MediaInfo? info;
        try {
          info = await VideoCompress.compressVideo(file.path,
              quality: VideoQuality.Res640x480Quality,
              deleteOrigin: false,
              duration: 60,
              includeAudio: true,
              frameRate: 24
          );
        } catch (e) {
          VideoCompress.cancelCompression();
        }
    if (info != null) {
      File compressedVideo = File(info.path!);
      return compressedVideo;
    } else {
      return file;
    }
  }
```

**Comment by Arley011** [comment]
@jpetro416, Nope. As far as I tested, cancelCompression() doesn't actually stop compression nor throw an error

**Comment by jpetro416** [comment]
Try removing `duration: ` and run the compressor code in a new project. I just tested again and was able to see it cancel when I went back a screen (I ran `cancelCompression();` in my `onDispose()`)
